### DISCUSSION TITLE
新版本视频直链在 iOS 版本的 VLC 播放器中无法作为网络串流打开

### DISCUSSION THREAD

**Discussion by opoet** [spencerwooo/onedrive-vercel-index#481] [General]
经测试最新版本复制视频直链后在 iOS 版本的 VLC 播放器内的网络串流打开会出现错误

![image](https://user-images.githubusercontent.com/30021379/154948790-90bf8c3e-088d-4ece-840d-d5c31d92cec9.png)

但是在老版本中复制的直链（带有参数 raw=true）是可以直接作为网络串流播放的，以下为新旧版本直链

 - https://dir.opoet.cn/api/raw/?path=/🛠️%20Test/300628168-1-208.mp4
 - https://dir.2wb.xyz/api?path=/300628168-1-208.mp4&raw=true

经过我多次的测试，新版本直链在 Android、Windows 端的 VLC 播放器中均没有此问题，可以作为网络串流播放，仅在 iOS 版本的 VLC 中出现此问题

**在 iOS 端浏览器中点击视频页面下方的 `VLC` 按钮跳转到 VLC 中可以正常播放**

**Comment by myl7** [comment]
我没有 IOS 设备没法做测试，你试试 <https://share.myl.moe/api/raw/?path=/test/测试一下.mp4> 和 <https://share.myl.moe/api/raw/?path=/test/%E6%B5%8B%E8%AF%95%E4%B8%80%E4%B8%8B.mp4> 这两个 URL 看看 IOS VLC 能否正确处理

**Reply by opoet** [reply]
第一个不行，第二个可以

**Reply by myl7** [reply]
那就是 URL encoding 的问题了，应该需要给 `自定义直链` 加一个可选项复制 URL encode 过的 URL。我待会儿看看。

**Reply by opoet** [reply]
感谢 🌹
